File: TimeTec/app/src/main/java/com/time/timetec/Activities/LoginActivity.java
package com.time.timetec.Activities;

import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;

import android.app.ProgressDialog;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;

import com.time.timetec.R;
import com.time.timetec.Retrofit.ApiClient;
import com.time.timetec.Models.LoginInfo;
import com.time.timetec.SharedPrefManager;

import java.util.ArrayList;
import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class LoginActivity extends AppCompatActivity {

    private static final String TAG = "Login";
    private Context mContext;

    ImageView top_curve,logo;
    EditText email,password;
    TextView email_text, password_text;
    LinearLayout new_user_layout;
    CardView login_card;
    List<LoginInfo> loginInfos;


    private ProgressDialog progressDialog;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        top_curve = findViewById(R.id.top_curve);
        email = findViewById(R.id.email);
        email_text = findViewById(R.id.email_text);
        password = findViewById(R.id.password);
        password_text = findViewById(R.id.password_text);
        logo = findViewById(R.id.logo);
        new_user_layout = findViewById(R.id.new_user_text);
        login_card = findViewById(R.id.login_card);
        loginInfos=new ArrayList<>();

        progressDialog =new ProgressDialog(this);


        Animation top_curve_anim = AnimationUtils.loadAnimation(getApplicationContext(),R.anim.top_down);
        top_curve.startAnimation(top_curve_anim);

        Animation editText_anim = AnimationUtils.loadAnimation(getApplicationContext(),R.anim.edittext_anim);
        email.startAnimation(editText_anim);
        password.startAnimation(editText_anim);

        Animation field_name_anim = AnimationUtils.loadAnimation(getApplicationContext(),R.anim.field_name_anim);
        email_text.startAnimation(field_name_anim);
        password_text.startAnimation(field_name_anim);
        logo.startAnimation(field_name_anim);

        Animation center_reveal_anim = AnimationUtils.loadAnimation(getApplicationContext(),R.anim.center_reveal_anim);
        login_card.startAnimation(center_reveal_anim);

        Animation new_user_anim = AnimationUtils.loadAnimation(getApplicationContext(),R.anim.down_top);
        new_user_layout.startAnimation(new_user_anim);



    }

    public void register(View view) {
        startActivity(new Intent(LoginActivity.this,Registration.class));

    }

    public void loginButton(View view) {
        // Toast.makeText(this,"Login Clicked",Toast.LENGTH_SHORT).show();
//        startActivity(new Intent(LoginActivity.this,MainActivity.class));


        String email_text=email.getText().toString().trim();
        String passord_text=password.getText().toString().trim();
        login(email_text,passord_text);

    }



    private void login(String email,String password){
        Call<List<LoginInfo>> call= ApiClient.getmInstance().getApi().login(
            email,password
        );



        call.enqueue(new Callback<List<LoginInfo>>() {
            @Override
            public void onResponse(Call<List<LoginInfo>> call, Response<List<LoginInfo>> response) {

                loginInfos=response.body();
                Log.d(TAG, "onResponse: list size : "+loginInfos.size());
                Log.d(TAG, "onResponse: list size : "+loginInfos.get(0).getLoginSuccess());
                Log.d(TAG, "onResponse: list size : "+loginInfos.get(0).getCustomerId());

                if (loginInfos.get(0).getLoginSuccess().equals("1")){
                    Toast.makeText(LoginActivity.this, "Login Successful", Toast.LENGTH_SHORT).show();

                    SharedPrefManager.getInstance(LoginActivity.this).saveUser(loginInfos.get(0).getCustomerId());
                    startActivity(new Intent(LoginActivity.this,MainActivity.class));
                    finish();

                }

                else {
                    Toast.makeText(LoginActivity.this, "Wrong User Name And Password", Toast.LENGTH_SHORT).show();
                }
            }

            @Override
            public void onFailure(Call<List<LoginInfo>> call, Throwable t) {
                Toast.makeText(LoginActivity.this, ""+t.getMessage(), Toast.LENGTH_SHORT).show();

            }
        });
    }

}

File: PreviousProjects/Zhomprass/app/src/main/java/com/zhomprass/zhomprasslimited/Activities/LoginActivity.kt
package com.zhomprass.zhomprasslimited.Activities

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.view.animation.AnimationUtils
import android.widget.*
import com.scwang.wave.MultiWaveHeader
import com.zhomprass.zhomprasslimited.ApiClient.ApiClient
import com.zhomprass.zhomprasslimited.Models.Members
import com.zhomprass.zhomprasslimited.Models.UserInfo
import com.zhomprass.zhomprasslimited.R
import com.zhomprass.zhomprasslimited.Utlis.SharedPrefManager
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class LoginActivity : AppCompatActivity() {

    var top_curve: ImageView? = null
    var email: EditText? = null
    var password: EditText? = null
    var zipMember: TextView? = null
    var freeMember: TextView? = null
    var logo1: TextView? = null
    var logo2: TextView? = null
    var logo: TextView? = null
    lateinit var login_button:Button
    lateinit var multiWaveeHeader:ImageView
    lateinit var progressBar: ProgressBar

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.login)

        top_curve = findViewById(R.id.top_curve)
        email = findViewById(R.id.ed_name)
        password = findViewById(R.id.ed_pass)
        logo = findViewById(R.id.logo)
        freeMember = findViewById(R.id.new_user_text)
        zipMember = findViewById(R.id.tv_or)
        login_button = findViewById(R.id.nxt)
        logo1 = findViewById(R.id.logo1)
        logo2 = findViewById(R.id.logo2)
        multiWaveeHeader=findViewById(R.id.waveHeader)
        progressBar=findViewById(R.id.progressBar)
        animation()


//        multiWaveeHeader.velocity= 1.0F
//        multiWaveeHeader.progress=1.0F
//        multiWaveeHeader.isRunning
//        multiWaveeHeader.gradientAngle=45
//        multiWaveeHeader.waveHeight=50
//        multiWaveeHeader.startColor=resources.getColor(R.color.orange)
//        multiWaveeHeader.closeColor=resources.getColor(R.color.Aqua)


        getMemberZpl()
        getMemberFree()


        login_button.setOnClickListener(View.OnClickListener {
            login()
        })


    }


    private fun animation() {
        val top_curve_anim = AnimationUtils.loadAnimation(applicationContext, R.anim.top_down)
        top_curve!!.startAnimation(top_curve_anim)
        val editText_anim = AnimationUtils.loadAnimation(applicationContext, R.anim.edittext_anim)
        email!!.startAnimation(editText_anim)
        password!!.startAnimation(editText_anim)
        val field_name_anim = AnimationUtils.loadAnimation(applicationContext, R.anim.field_name_anim)
        logo!!.startAnimation(field_name_anim)
        val center_reveal_anim = AnimationUtils.loadAnimation(applicationContext, R.anim.center_reveal_anim)
        login_button!!.startAnimation(center_reveal_anim)
        val freeMember_anim = AnimationUtils.loadAnimation(applicationContext, R.anim.down_top)
        freeMember!!.startAnimation(freeMember_anim)
        val zipMember_anim = AnimationUtils.loadAnimation(applicationContext, R.anim.down_top)
        zipMember!!.startAnimation(zipMember_anim)
        val logo1_anim = AnimationUtils.loadAnimation(applicationContext, R.anim.down_top)
        logo1!!.startAnimation(logo1_anim)
        val logo2_anim = AnimationUtils.loadAnimation(applicationContext, R.anim.down_top)
        logo2!!.startAnimation(logo2_anim)
    }




    private fun getMemberZpl(){
        ApiClient.instance.getTotalZpl("zpl")
            .enqueue(object :Callback<List<Members>>{
                override fun onFailure(call: Call<List<Members>>, t: Throwable) {


                }

                override fun onResponse(
                    call: Call<List<Members>>,
                    response: Response<List<Members>>
                ) {
                    if (response.isSuccessful){
                        if (response.body()==null){
                            return
                        }

                        else{
                            logo1!!.text="${response.body()!!.get(0).member}"
                        }
                    }
                }

            })
    }



    private fun getMemberFree(){
        ApiClient.instance.getTotalZpl("free")
            .enqueue(object :Callback<List<Members>>{
                override fun onFailure(call: Call<List<Members>>, t: Throwable) {


                }

                override fun onResponse(
                    call: Call<List<Members>>,
                    response: Response<List<Members>>
                ) {
                    if (response.isSuccessful){
                        if (response.body()==null){
                            return
                        }

                        else{
                            logo2!!.text="${response.body()!!.get(0).member}"
                        }
                    }
                }

            })
    }


    private fun login(){

        if (validation()){

            progressBar.visibility=View.VISIBLE
            var username= email?.text.toString();
            var pass:String= password?.text.toString();

            ApiClient.instance.login(
                username,
                pass
            ).enqueue(object :Callback<List<UserInfo>>{
                override fun onFailure(call: Call<List<UserInfo>>, t: Throwable) {

                }

                override fun onResponse(
                    call: Call<List<UserInfo>>,
                    response: Response<List<UserInfo>>
                ) {
                    if (response.isSuccessful){


                      val userInfo=response.body()
                        println(userInfo!!.get(0).login_success)
                        if (userInfo.get(0).login_success==1){
                            SharedPrefManager.getInstance(this@LoginActivity)?.saveUser(userInfo.get(0).customer_id)
                            progressBar.visibility=View.GONE

                            Toast.makeText(this@LoginActivity,"Login Successful",Toast.LENGTH_SHORT).show()
                           startActivity(Intent(this@LoginActivity,MainActivity::class.java))
                            finish()
                        }

                        else{
                            progressBar.visibility=View.GONE

                            Toast.makeText(this@LoginActivity,"Wrong Username And Password",Toast.LENGTH_SHORT).show()

                        }
                    }
                }

            })
        }
    }



    private fun validation():Boolean{
        if (email!!.text.isNullOrEmpty()){
            email!!.error="Enter a username"
            email!!.requestFocus()

            return false

        }

        if (password!!.text.isNullOrEmpty()){
            password!!.error="Enter a password"
            password!!.requestFocus()

            return false

        }

        return true

    }
}
